### PR TITLE
fix(plugins): add bearer token to request headers when sending requests, authmiddleware issue

### PR DESCRIPTION
**Pull request by pbuehl-redhat**
…iddleware issue

### CODE REVIEW

File: plugins/quay/src/api/index.ts [hunk 1]
```diff
@@ -60,8 +65,12 @@ export class QuayApiClient implements QuayApiV1 {
   }
 
   private async fetcher(url: string) {
+    const {token: idToken} = await this.identityApi.getCredentials();
     const response = await fetch(url, {
-      headers: { 'Content-Type': 'application/json' },
+      headers: {
+        'Content-Type': 'application/json',
+        ...( idToken && {'Authorization': `Bearer ${idToken}`})
```
Comment (invincibleJai) [inline plugins/quay/src/api/index.ts:72]: is this needed, as with proxy header are propagated automatically and won't be seen on the client side. This plugin I have verified and similar should be case with Jfrog as well https://backstage.io/docs/tutorials/using-backstage-proxy-within-plugin/

**Comment by invincibleJai** [comment]
@pbuehl-redhat Thanks for your contribution, would recommend linking the issue on the ticket.

cc @tumido

**Comment by pbuehl-redhat** [comment]
@invincibleJai Thank you! I'm not exactly sure how to link the issue, but this is the issue that it needs to be linked with https://github.com/janus-idp/backstage-plugins/issues/486

**Comment by invincibleJai** [comment]
> #486

@pbuehl-redhat you can add this to the PR description to start with, i.e Fixes: https://github.com/janus-idp/backstage-plugins/issues/486

and the bug you shared talks about `quay backstage plugins` but PR changes are made to more plugins like Jfrog, OCM, and Quay

cc @fmenesesg @christophe-f @tumido  ^^

**Review by tumido — COMMENTED** [review]
CI is complaining, can you please run `yarn prettier:fix` on this PR?

Can you please also take a look at the https://github.com/janus-idp/backstage-plugins/pull/498#discussion_r1250390232 comment?

**Comment by malagant** [comment]
I'm absolutely in need of this PR to be merged. Any hints, on when this will happen?

**Comment by tumido** [comment]
Hi @malagant this PR has still 1 unresolved discussion that is awaiting a reply from from the PR author. Maybe you can help there?

Can you help me understand for which plugin do you need this for the most?

I'm also afraid that since all except OCM plugin affected by this PR here rely on Backstage proxy, this change actually doesn't change anything for them. As far as I know, identity api is not used by Backstage proxy.

File: plugins/quay/src/api/index.ts (hunk 1 above)
Comment (tumido) [reply plugins/quay/src/api/index.ts]: @pbuehl-redhat it seems to me like you're trying to implement https://backstage.io/docs/auth/service-to-service-auth#usage-in-backend-plugins for plugins which **don't** have a backend plugin counterpart.

**Comment by malagant** [comment]
Hi @tumido, thanks for the really fast response. I am using the quay plugin. Wanted to showcase it but found out that backend auth of backstage is not implemented. I have a workaround: Disable backend auth. But this only fits because we don't expose anything outside the company.

Let me check, what's blocking and jump into the discussion. Would be glad to support/help.

**Comment by tumido** [comment]
Interesting.... @malagant can you please share with me your backend app configuration for the quay plugin? What confuses me is that Quay is using Backstage backend proxy...

I suppose you use something like:

```ts
  apiRouter.use('/proxy', authMiddleware, await proxy(proxyEnv));
```

correct?

File: plugins/quay/src/api/index.ts (hunk 1 above)
Comment (tumido) [reply plugins/quay/src/api/index.ts]: @invincibleJai I think I'm finding the answers myself. You can run the proxy with auth middleware. In that case you need to be able to authenticate against it.

**Comment by malagant** [comment]
> Interesting.... @malagant can you please share with me your backend app configuration for the quay plugin? What confuses me is that Quay is using Backstage backend proxy...
> 
> I suppose you use something like:
> 
> ```ts
>   apiRouter.use('/proxy', authMiddleware, await proxy(proxyEnv));
> ```
> 
> correct?
```
proxy:
  # Quay
  '/quay/api':
    target: 'https://...'
    headers:
      X-Requested-With: 'XMLHttpRequest'
      Authorization: 'Bearer ${MTR_TOKEN}'
``` 

And yes you're correct.

**Comment by tumido** [comment]
I that case it all makes sense to me. 🙂 Let's get this merged. 👍

File: plugins/quay/src/api/index.ts (hunk 1 above)
Comment (tumido) [reply plugins/quay/src/api/index.ts]: However, as I've learnt now, you can make Proxy to use backend service auth as well. In that case this applies to the  plugins which use proxy as well.

```ts
  apiRouter.use('/proxy', authMiddleware, await proxy(proxyEnv));
```

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**